### PR TITLE
refactor: improve logging

### PR DESCRIPTION
**Pull request by subotic**
resolves DEV-

### CODE REVIEW

File: apps/dsp-app/src/app/main/services/datadog-rum.service.ts [hunk 1]
```diff
@@ -1,63 +1,67 @@
...
-            // if session is valid: setActiveUser
-            this._s.isSessionValid().subscribe((response: boolean) => {
-                if (response) {
-                    const session: Session = this._s.getSession();
-                    this.setActiveUser(session.user.name, 'username');
-                }
-            });
-        }
+                // if session is valid: setActiveUser
+                this.session.isSessionValid().subscribe((response: boolean) => {
```
Comment (Vijeinath) [inline apps/dsp-app/src/app/main/services/datadog-rum.service.ts:57]: Where do you unsubscribe?

File: libs/vre/shared/app-config/src/lib/build-tag/build-tag.ts [hunk 2]
```diff
@@ -0,0 +1,24 @@
+/*
+ * Copyright © 2021 - 2023 Swiss National Data and Service Center for the Humanities and/or DaSCH Service Platform contributors.
+ *  SPDX-License-Identifier: Apache-2.0
+ */
+
+import { z } from 'zod';
+import { inject } from '@angular/core';
+import { HttpClient } from '@angular/common/http';
+import { tap, map } from 'rxjs/operators';
```
Comment (Vijeinath) [inline libs/vre/shared/app-config/src/lib/build-tag/build-tag.ts:9]: Where do you use "tap"?

File: libs/vre/shared/app-logging/src/lib/app-logging/app-logging.service.ts [hunk 3]
```diff
@@ -1,31 +1,45 @@
...
-                useSecureSessionCookie: true,
-            });
-            datadogLogs.logger.setHandler(['console', 'http']);
-            this.logger = datadogLogs.logger;
-        }
+    private buildTag$: Observable<BuildTag> = inject(BuildTagToken);
+    private config: DspInstrumentationConfig = inject(DspInstrumentationToken);
+
+    constructor() {
+        this.buildTag$.pipe(first()).subscribe((tag) => {
```
Comment (Vijeinath) [inline libs/vre/shared/app-logging/src/lib/app-logging/app-logging.service.ts:22]: Where do you unsubscribe?

**Review by Vijeinath — APPROVED** [review]
Small stuffs

File: apps/dsp-app/src/app/main/services/datadog-rum.service.ts [hunk 4]
```diff
@@ -1,63 +1,67 @@
...
-                            ).response?.body,
-                        };
-                    }
-                },
-            });
+    private buildTag$: Observable<BuildTag> = inject(BuildTagToken);
+    private config: DspInstrumentationConfig = inject(DspInstrumentationToken);
+    private session = inject(SessionService);
+    constructor() {
+        this.buildTag$.pipe(first()).subscribe((tag) => {
```
[the remark this answers is not in the document]
Comment (subotic) [reply apps/dsp-app/src/app/main/services/datadog-rum.service.ts:24]: first does it automatically after the first value is received.

File: apps/dsp-app/src/app/main/services/datadog-rum.service.ts (hunk 1 above)
Comment (subotic) [reply apps/dsp-app/src/app/main/services/datadog-rum.service.ts:57]: nowhere. This is already existing code and out of scope for this round.

File: libs/vre/shared/app-config/src/lib/build-tag/build-tag.ts (hunk 2 above)
Comment (subotic) [reply libs/vre/shared/app-config/src/lib/build-tag/build-tag.ts:9]: nowhere. will remove it.

File: libs/vre/shared/app-logging/src/lib/app-logging/app-logging.service.ts (hunk 3 above)
Comment (subotic) [reply libs/vre/shared/app-logging/src/lib/app-logging/app-logging.service.ts:22]: first()

**Comment by subotic** [comment]
Thanks for the review!

**Pull request merged**